packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,317 @@
+    );
+  }
+
+  // Step 3: Create a new transaction template for the child transaction
+  const childTxTemplate = new BtcTransactionTemplate({
+    inputs: [],
+    outputs: [],
+    targetFeeRate: Number(txAnalyzer.cpfpFeeRate), // Use CPFP-specific fee rate from analyzer
+    dustThreshold,
+    network,
+    scriptType,
+    requiredSigners,
+    totalSigners,
+  });
+
+  // Step 4: Add the spendable output from the parent transaction as an input
+  const parentOutput = txAnalyzer.outputs[spendableOutputIndex];
+  if (!parentOutput) {
+    throw new Error(
+      `Spendable output at index ${spendableOutputIndex} not found in the parent transaction.`,
+    );
+  }
+
+  // Create a UTXO from the parent transaction's output
+  const parentOutputUTXO: UTXO = {
+    txid: analysis.txid,
+    vout: spendableOutputIndex,
+    value: parentOutput.amountSats,
+    witnessUtxo: {
+      script: createOutputScript(parentOutput.address, network),
+      value: parseInt(parentOutput.amountSats),
+    },
+
+    prevTxHex: originalTx,
+  };
+
+  childTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(parentOutputUTXO));
+
+  // Step 5: Add a change output (at least 1 output required for a valid transaction)
+  childTxTemplate.addOutput(
+    new BtcTxOutputTemplate({
+      address: changeAddress,
+      amountSats: "0", // Initial amount, will be adjusted later
+    }),
+  );
+
+  /**
+   * Step 6: Add additional inputs to cover the CPFP fee requirements
+   *
+   * This step implements a simplified version of the "Child-pays-for-parent" (CPFP)
+   * transaction selection logic, inspired by Bitcoin Core's implementation
+   * (https://github.com/bitcoin/bitcoin/pull/7600).
+   *
+   * In Bitcoin Core, transactions are sorted in the mempool using a modified
+   * feerate that considers ancestors. During block creation, transactions are
+   * considered in this order. As transactions are selected, a separate map tracks
+   * the new feerate-with-ancestors for in-mempool descendants.
+   *
+   * Our simplified approach:
+   * 1. We start with the parent transaction's fee and vsize.
+   * 2. We iteratively add inputs to the child transaction.
+   * 3. After each input addition, we recalculate the combined fee rate of the
+   *    parent and child transactions.
+   * 4. We continue adding inputs until the combined fee rate meets or exceeds
+   *    the target fee rate.
+   *
+   * This method ensures that the child transaction provides enough fee to incentivize
+   * miners to include both the parent and child transactions in a block, effectively
+   * "bumping" the fee of the parent transaction.
+   *
+   * Note: While this approach is simpler than Bitcoin Core's full implementation,
+   * it achieves the core goal of CPFP: ensuring the combined package (parent + child)
+   * is attractive for miners to include in a block.
+   */
+
+  for (const utxo of availableInputs) {
+    if (isCPFPFeeSatisfied(txAnalyzer, childTxTemplate)) {
```

Comment:
I think you need a check for the dust buffer here. you want to make sure to stop adding inputs once the fee is satisfied and you have enough output value to not have dust.

Reply:
changes done : [review-changes: CPFP](https://github.com/caravan-bitcoin/caravan/pull/114/commits/40576591fc50b736bdf91485b2e02eedbd2feb97)